test/Test_Evaluator.re

```diff
@@ -413,6 +413,46 @@ let test_typfun_application = () =>
     |> Exp.fresh,
   );
 
+let qcheck_evaluator_does_not_crash_test =
+  QCheck.Test.make(
+    ~name="Evaluator does not crash", ~count=100000, QCheck_Util.arb_exp, exp => {
+    switch (
+      Evaluator.evaluate(
+        ~env=Builtins.env_init,
+        ~step_limit=10000,
+        elaborate(exp),
+      )
+    ) {
+    | (_, _) => true
+    | exception e =>
+      print_endline(
+        "ExpToSegment: "
+        ++ Printer.of_segment(
+             ~holes=Some("?"),
+             ExpToSegment.exp_to_segment(
+               ~settings={
+                 inline: true,
+                 fold_case_clauses: false,
+                 fold_fn_bodies: false,
+                 hide_fixpoints: false,
+                 fold_cast_types: false,
+                 show_filters: true,
+                 show_unknown_as_hole: true,
+               },
+               exp,
+             ),
+           ),
+      );
```

Comment:
We can stop printing this. Maybe we print every x step limit reached or something.